native-plasma/app/src/main/jni/plasma_renderer_rgb565.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2015 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "plasma_renderer.h"
+
+// Set to 1 to optimize memory stores when generating plasma.
+#define OPTIMIZE_WRITES 1
+
+/* Color palette used for rendering the plasma */
+static const int PALETTE_BITS = 8;
+static const int PALETTE_SIZE = 1 << PALETTE_BITS;
+
```

Comment:
this palette things are for 565 format, not related to 32 bit (A888) format, correct? wondering it could get pushed and tucked inside only 565 format implementation. 

probably original person declare everything in one place to allow for inline so it is faster?  if we pushed it inside the class, it might get impact on performance ( this is for every pixel ). 

I like the way you think to tuck them into a class